rust/operator-binary/src/crd/mod.rs

```diff
@@ -111,184 +115,274 @@ pub enum Error {
+        #[fragment_attrs(serde(default))]
+        pub requested_secret_lifetime: Option<Duration>,
+    }
 
-#[derive(Clone, Debug, Default, JsonSchema, PartialEq, Fragment)]
-#[fragment_attrs(
-    derive(
+    #[derive(Clone, Debug, Default, JsonSchema, PartialEq, Fragment)]
+    #[fragment_attrs(
+        derive(
+            Clone,
+            Debug,
+            Default,
+            Deserialize,
+            Merge,
+            JsonSchema,
+            PartialEq,
+            Serialize
+        ),
+        serde(rename_all = "camelCase")
+    )]
+    pub struct ZookeeperStorageConfig {
+        #[fragment_attrs(serde(default))]
+        pub data: PvcConfig,
+    }
+
+    #[derive(
         Clone,
         Debug,
-        Default,
         Deserialize,
-        Merge,
+        Display,
+        Eq,
+        EnumIter,
         JsonSchema,
+        Ord,
         PartialEq,
-        Serialize
-    ),
-    serde(rename_all = "camelCase")
-)]
-pub struct ZookeeperStorageConfig {
-    #[fragment_attrs(serde(default))]
-    pub data: PvcConfig,
+        PartialOrd,
+        Serialize,
+    )]
+    #[serde(rename_all = "camelCase")]
+    pub enum Container {
+        Prepare,
+        Vector,
+        Zookeeper,
+    }
+
+    #[derive(Clone, Default, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct ZookeeperClusterStatus {
+        /// An opaque value that changes every time a discovery detail does
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        pub discovery_hash: Option<String>,
+        #[serde(default)]
+        pub conditions: Vec<ClusterCondition>,
+    }
+
+    /// A claim for a single ZooKeeper ZNode tree (filesystem node).
+    ///
+    /// A ConfigMap will automatically be created with the same name, containing the connection string in the field `ZOOKEEPER`.
+    /// Each ZookeeperZnode gets an isolated ZNode chroot, which the `ZOOKEEPER` automatically contains.
+    /// All data inside of this chroot will be deleted when the corresponding `ZookeeperZnode` is.
+    ///
+    /// `ZookeeperZnode` is *not* designed to manage the contents of this ZNode. Instead, it should be used to create a chroot
+    /// for an installation of an application to work inside. Initializing the contents is the responsibility of the application.
+    ///
+    /// You can learn more about this in the
+    /// [Isolating clients with ZNodes usage guide](DOCS_BASE_URL_PLACEHOLDER/zookeeper/usage_guide/isolating_clients_with_znodes).
+    #[derive(Clone, CustomResource, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+    #[versioned(k8s(
+        group = "zookeeper.stackable.tech",
+        kind = "ZookeeperZnode",
+        plural = "zookeeperznodes",
+        // TODO: stackable-versioned currently only supports a single shortname.
+        // Leaving this one commented for now.
+        // shortname = "zno",
```

Comment:
Multiple shortnames are now supported since stackable-versioned 0.5.1

```suggestion
        shortname = "zno",
```

Reply:
https://github.com/stackabletech/zookeeper-operator/pull/917/commits/a5cc9fb7bddc044e0958d26325234f569d2e7c78